protocol/bin-prot/src/polyvar/caml_hash_variant.rs

```diff
@@ -0,0 +1,45 @@
+// Copyright 2020 ChainSafe Systems
+// SPDX-License-Identifier: Apache-2.0
+
+//! OCaml polyvars use a special integer output hash function to index variants
+//! This module reimplements this hash function
+
+/// Type alias for hash result
+pub type VariantHash = u32;
+
+/// Hash the label string (ASCII not UTF-8) into a OCaml style variant hash
+pub fn caml_hash_variant(label: &[u8]) -> VariantHash {
+    label
+        .iter()
+        .fold(0_u32, |accu, byte| 223 * accu + (*byte as u32))
+}
+
+/* original C code
+    CAMLexport value caml_hash_variant(char const * tag)
+    {
+      value accu;
+      /* Same hashing algorithm as in ../typing/btype.ml, function hash_variant */
+      for (accu = Val_int(0); *tag != 0; tag++)
+        accu = Val_int(223 * Int_val(accu) + *((unsigned char *) tag));
+    #ifdef ARCH_SIXTYFOUR
+      accu = accu & Val_long(0x7FFFFFFFL);
```

Comment:
It looks like the intermediate `accu` is `i64`, is it safe to `u32` in rust port? Maybe add some test cases where `u32` overflow on add happens